Wikipedia/Code/TalkPageLocalHandler.swift

```diff
@@ -0,0 +1,97 @@
+
+import Foundation
+
+class TalkPageLocalHandler {
+    
+    var dataStore: MWKDataStore
+    
+    required init(dataStore: MWKDataStore) {
+        self.dataStore = dataStore
+    }
+    
+    func existingTalkPage(for taskURL: URL) throws -> TalkPage? {
+        
+        guard let databaseKey = taskURL.wmf_talkPageDatabaseKey else {
+            throw TalkPageError.talkPageDatabaseKeyCreationFailure
+        }
+        
+        let fetchRequest: NSFetchRequest<TalkPage> = TalkPage.fetchRequest()
+        fetchRequest.fetchLimit = 1
+        fetchRequest.predicate = NSPredicate(format: "key == %@", databaseKey)
+        
+        return try dataStore.viewContext.fetch(fetchRequest).first
+    }
+    
+    func updateExistingTalkPage(existingTalkPage: TalkPage, with networkTalkPage: NetworkTalkPage) -> TalkPage? {
+        existingTalkPage.name = networkTalkPage.name
+        existingTalkPage.revisionId = networkTalkPage.revisionId
+        
+        guard let newDiscussions: [TalkPageDiscussion] = createTalkPageDiscussions(with: networkTalkPage) else {
+            return nil
+        }
+        
+        //first delete old discussions
+        //todo: is there a better way to do this? simply reassigning discussions is not enough
```

Comment:
If I'm understanding this correctly, you could compare your existing discussions with the discussions that are coming from the server and figure out what changed (what was deleted, what was added, what was updated) and then process those specific changes - we do something like that for reading lists -https://github.com/wikimedia/wikipedia-ios/blob/908bd09ae9e8c0907488512b149d5ce049c5a48d/WMF%20Framework/ReadingListsSyncOperation.swift#L244

specifically here's where the updates happen - https://github.com/wikimedia/wikipedia-ios/blob/908bd09ae9e8c0907488512b149d5ce049c5a48d/WMF%20Framework/ReadingListsSyncOperation.swift#L290

Reply:
Oh yeah that would be better. Right now I'm just throwing out all discussions and recreating them if the revision ID has changed. The question was more around delete rules - the old discussions still hang around in the DB even after I reassign the talk page to a new set of discussions.  This was a note to look into a way to have that delete cascade on that assignment without explicitly looping through and deleting them all. But agreed being more selective about what I delete would be a better way to go.